src/main/java/fi/dy/masa/servux/dataproviders/StructureDataProvider.java

```diff
@@ -404,4 +494,76 @@ protected NbtList getStructureList(Map<ChunkPos, StructureStart> structures, Ser
 
         return list;
     }
+
+    // TODO --> Move out of structures channel in the future (Some Metadata channel, perhaps)
+    public void refreshSpawnMetadata(ServerPlayerEntity player, @Nullable NbtCompound data)
+    {
+        NbtCompound nbt = new NbtCompound();
+        BlockPos spawnPos = StructureDataProvider.INSTANCE.getSpawnPos();
+
+        nbt.putInt("packetType", ServuxStructuresHandler.PACKET_S2C_SPAWN_METADATA);
+        nbt.putString("id", getNetworkChannel().toString());
+        nbt.putString("servux", Reference.MOD_STRING);
+        nbt.putInt("spawnPosX", spawnPos.getX());
+        nbt.putInt("spawnPosY", spawnPos.getY());
+        nbt.putInt("spawnPosZ", spawnPos.getZ());
+        nbt.putInt("spawnChunkRadius", StructureDataProvider.INSTANCE.getSpawnChunkRadius());
+
+        HANDLER.encodeNbtCompound(player, nbt);
+    }
+
+    public BlockPos getSpawnPos()
+    {
+        if (this.spawnPos == null)
+        {
+            this.setSpawnPos(BlockPos.ORIGIN);
+        }
+
+        return this.spawnPos;
+    }
+
+    public void setSpawnPos(BlockPos spawnPos)
+    {
+        if (this.spawnPos.equals(spawnPos) == false)
+        {
+            this.metadata.remove("spawnPosX");
+            this.metadata.remove("spawnPosY");
+            this.metadata.remove("spawnPosZ");
+            this.metadata.putInt("spawnPosX", spawnPos.getX());
+            this.metadata.putInt("spawnPosY", spawnPos.getY());
+            this.metadata.putInt("spawnPosZ", spawnPos.getZ());
+            this.refreshSpawnMetadata = true;
+
+            //Servux.logger.info("setSpawnPos(): updating World Spawn [{}] -> [{}]", this.spawnPos.toShortString(), spawnPos.toShortString());
+        }
+
+        this.spawnPos = spawnPos;
+    }
+
+    public int getSpawnChunkRadius()
+    {
+        if (this.spawnChunkRadius < 0)
+        {
+            this.spawnChunkRadius = 2;
+        }
+
+        return this.spawnChunkRadius;
+    }
+
+    public void setSpawnChunkRadius(int radius)
+    {
+        if (this.spawnChunkRadius != radius)
+        {
+            this.metadata.remove("spawnChunkRadius");
+            this.metadata.putInt("spawnChunkRadius", radius);
+            this.refreshSpawnMetadata = true;
+
+            //Servux.logger.info("setSpawnPos(): updating Spawn Chunk Radius [{}] -> [{}]", this.spawnChunkRadius, radius);
```

Comment:
I believe Log4j has the `#debug()` method, maybe use that one an uncomment the lines?

Reply:
No, these are development debug lines and could be deleted, not uncommented.